### PR TITLE
Remove rxjs compat

### PR DESCRIPTION
**Pull request by chrste90**
This PR changes the rxjs-imports to the new structure and removes rxjs-compat.

### CODE REVIEW

File: src/mqtt.service.ts [hunk 1]
```diff
@@ -3,14 +3,7 @@ import { ISubscriptionGrant } from './mqtt-types';
 import { connect } from '../vendor/mqtt.browserified.js';
 import * as extend from 'xtend';
```
Comment (sclausen) [inline src/mqtt.service.ts]: Isn't importing everything directly from `rxjs` blowing up the size of the bundle, since it's not tree shakeable anymore?

Comment (chrste90) [reply src/mqtt.service.ts]: No not anymore with the cli v6 (together with webpack 4). Rxjs v6 is marked as sideEffect-free in package.json which allows webpack to remove all the code which is not used. You can read more about it here: https://github.com/webpack/webpack/tree/master/examples/side-effects

The rxjs-compat package only exist to allow a smooth migration to the new api of rxjs and is not tree-shakable. Here are some more information: https://github.com/ReactiveX/rxjs/blob/master/MIGRATION.md

**Comment by chrste90** [comment]
Should be fixed now. I forgot to change the import in `spec-bundle.js`.

**Comment by chrste90** [comment]
And i needed to make some changes to the tests because i got timeouts on travis.

File: tests/mqtt.service.spec.ts [hunk 2]
```diff
@@ -10,37 +10,41 @@ const config: IMqttServiceOptions = {
...
+        {
+          provide: MqttClientService,
+          useValue: undefined
+        }
+      ]
+    });
+    mqttService = TestBed.get(MqttService);
+  });
+
+  afterEach(function () {
```
[the remark this answers is not in the document]
Comment (chrste90) [reply tests/mqtt.service.spec.ts]: Done

**Comment by sclausen** [comment]
@chrste90 Just for your information: Replacing the functionality of `done` with `async` caused the tests to pass even if they should fail (no running broker for example).

**Pull request merged**